specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2019-02-01-preview/EventGrid.json

```diff
@@ -0,0 +1,3752 @@
+                "consumes": [
+                    "application/json"
+                ],
+                "produces": [
+                    "application/json"
+                ],
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "name": "resourceGroupName",
+                        "in": "path",
+                        "description": "The name of the resource group within the user's subscription.",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "domainName",
+                        "in": "path",
+                        "description": "Name of the domain",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "domainUpdateParameters",
+                        "in": "body",
+                        "description": "Domain update information",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/DomainUpdateParameters"
+                        }
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "201": {
+                        "description": "Domain update request accepted.",
+                        "schema": {
+                            "$ref": "#/definitions/Domain"
+                        }
+                    },
+                    "default": {
+                        "description": "*** Error Responses: ***\n\n * 400 Bad Request.\n\n * 500 Internal Server Error."
+                    }
+                },
+                "x-ms-examples": {
+                    "Domains_Update": {
+                        "$ref": "./examples/Domains_Update.json"
+                    }
+                },
+                "x-ms-long-running-operation": true
+            }
+        },
+        "/subscriptions/{subscriptionId}/providers/Microsoft.EventGrid/domains": {
+            "get": {
+                "tags": [
+                    "Domains"
+                ],
+                "summary": "List domains under an Azure subscription",
+                "description": "List all the domains under an Azure subscription",
+                "operationId": "Domains_ListBySubscription",
+                "consumes": [],
+                "produces": [
+                    "application/json"
+                ],
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "name": "$filter",
```

Comment:
Please consider defining both these new parameters as global parameters (similar to other global parameters defined in the parameters section at the end of this swagger), this will avoid repeating the definition of these values in each list operation and will help the maintainability/future updates.

Reply:
Updated. thx for the suggestion